infra/src/main/kotlin/fi/fta/geoviite/infra/configuration/CachePreloadScheduler.kt

```diff
@@ -0,0 +1,39 @@
+package fi.fta.geoviite.infra.configuration
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Component
+
+@Component
+@ConditionalOnProperty(
+    name = ["geoviite.cache.enabled", "geoviite.cache.tasks.preload.enabled"],
+    havingValue = "true",
```

Comment:
Hieman outoa Springin toimintalogiikkaa: name on array, mutta näitä voidaan mätsätä ainoastaan yhteen arvoon.